### DISCUSSION TITLE
Getting user ID to automatically assign author of a record

### DISCUSSION THREAD

**Discussion by bictorman** [pocketbase/pocketbase#1374] [Q&A] [answered]
Hi,

First of all, fantastic job! I hope you can keep working on it.

I'm trying to find the best way to automatically assign the user that creates a record or changes it to a column, so it can not be tampered with via API.  I saw your suggestions here #200 but there might be corner cases and I would prefer to keep API rules simple.

How do I get the ID of the current user within a hook? I found this answer: `e.HttpContext.Get("user").(*models.User)`  in #514 but models.User does no longer exist and I don't see anything that looks similar.

Side feature request:  Set default values for fields, additionally also be able to set defaults using filter syntax, so in this case one could use `@request.auth.id`

**Comment by ganigeorgiev** [comment, marked as answer]
The linked discussions refer to older APIs and things have changed a little since then. It is better to use the examples from the site documentation - https://pocketbase.io/docs/router/.

> How do I get the ID of the current user within a hook?

```go
app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
    // to get the authenticated admin:
    admin, _ := e.HttpContext.Get(apis.ContextAdminKey).(*models.Admin)
    if admin == nil {
        return apis.NewForbiddenError("Only admins can access this endpoint", nil)
    }

    // or to get the authenticated record:
    authRecord, _ := e.HttpContext.Get(apis.ContextAuthRecordKey).(*models.Record)
    if authRecord == nil {
        return apis.NewForbiddenError("Only auth records can access this endpoint", nil)
    }

    ...

    return nil
})
```

For more examples, please refer to https://pocketbase.io/docs/router/#get-logged-admin-or-app-user.



---

> Side feature request: Set default values for fields, additionally also be able to set defaults using filter syntax, so in this case one could use @request.auth.id

Default values are not planned since it is not clear when they will be applied (note that PocketBase doesn't support `null` as field value, with the exception of the `json` type field). You can set defaults client-side or server-side, for example:

```go
app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
	if e.Record.Collection().Name != "myCollection" {
		return nil // not myCollection 
	}

	// set default myField value
	if e.Record.GetString("myField") == "" {
		// note that we don't need to explicitly call app.Dao().SaveRecord(e.Record)
		// because the finalizer of the OnRecordBefore hook will do it for us
		e.Record.Set("myField", "example")
	}

	return nil
})
```

**Reply by ganigeorgiev** [reply]
_Side note: In the next release we'll have support for `isset` filter modifier which could be used to ensure that an update request doesn't come with changed field values, eg.:_

```js
// prevent changing the role field
@request.data.role.isset = false && ...
```

**Comment by bictorman** [comment]
Aha! great. I did not think about looking into the router documentation for this.

> Default values are not planned since it is not clear when they will be applied (note that PocketBase doesn't support null as field value, with the exception of the json type field).

I see, maybe you could enable it only for nonempty fields (on empty value). But I understand if you think that's a bit clunky.

Thanks a lot!
